core/server/auth/oauth.js

```diff
@@ -86,6 +86,10 @@ function exchangeAuthorizationCode(req, res, next) {
             }));
         }
 
+        if (req.brute && req.brute.reset) {
+            req.brute.reset();
```

Comment:
Something to keep in mind is that the client does have a rudimentary method in place to prevent a situation where having the admin app open in multiple tabs will hit the token refresh endpoint multiple times in short succession but it's not guaranteed, especially if the `access_token` has expired.